pl_examples/basic_examples/text_classification.py

```diff
@@ -0,0 +1,136 @@
+'''
+class LitTextSentiment(pl.LightningModule):
+    def __init__(self, vocab_size, embed_dim, num_class, learning_rate=1e-3):
+        super().__init__()
+        self.save_hyperparameters()
+        self.embedding = nn.EmbeddingBag(vocab_size, embed_dim, sparse=True)
+        self.fc = nn.Linear(embed_dim, num_class)
+        self.init_weights()
+
+    def init_weights(self):
+        initrange = 0.5
+        self.embedding.weight.data.uniform_(-initrange, initrange)
+        self.fc.weight.data.uniform_(-initrange, initrange)
+        self.fc.bias.data.zero_()
+
+    def forward(self, text, offsets):
+        embedded = self.embedding(text, offsets)
+        return self.fc(embedded)
+
+
+    def training_step(self, batch, batch_idx):
+        x, offset, y = batch
+        y_hat = self(x,offset)
+        loss = F.cross_entropy(y_hat, y)
+        return loss
+
+    def validation_step(self, batch, batch_idx):
+        x,offset,y = batch
+        y_hat = self(x,offset)
+        loss = F.cross_entropy(y_hat, y)
+        self.log('valid_loss', loss)
+
+    def test_step(self, batch, batch_idx):
+        x,offset,y = batch
+        y_hat = self(x,offset)
+        loss = F.cross_entropy(y_hat, y)
+        self.log('test_loss', loss)
+
+    def configure_optimizers(self):
+        return torch.optim.SGD(self.parameters(), lr=self.hparams.learning_rate)
+
+    @staticmethod
+    def add_model_specific_args(parent_parser):
+        parser = ArgumentParser(parents=[parent_parser], add_help=False)
+        parser.add_argument('--learning_rate', type=float, default=0.0001)
+        return parser
+
+
+
+def generate_batch(batch):
+        label = torch.tensor([entry[0] for entry in batch])
+        text = [entry[1] for entry in batch]
+        offsets = [0] + [len(entry) for entry in text]
+        offsets = torch.tensor(offsets[:-1]).cumsum(dim=0)
+        text = torch.cat(text)
+        return text, offsets, label
+
+def run():
+    pl.seed_everything(1234)
+
+    # ------------
+    # args
+    # ------------
+    parser = ArgumentParser()
+    parser.add_argument('--batch_size', default=32, type=int)
+    #parser.add_argument('--hidden_dim', type=int, default=128)
+    parser = pl.Trainer.add_argparse_args(parser)
+    parser = LitTextSentiment.add_model_specific_args(parser)
+    args = parser.parse_args()
+
+    # ------------
+    # data
+    # ------------
+
+    train_dataset, test_dataset = text_classification.DATASETS['AG_NEWS'](root='./.data', ngrams=NGRAMS, vocab=None)
+    X_train, X_val = random_split(train_dataset, [100000, 20000])
+
+    train_loader = DataLoader(X_train, batch_size=args.batch_size, collate_fn=generate_batch)
+    val_loader = DataLoader(X_val, batch_size=args.batch_size, collate_fn=generate_batch)
+    test_loader = DataLoader(test_dataset, batch_size=args.batch_size, collate_fn=generate_batch)
+
+    VOCAB_SIZE = len(train_dataset.get_vocab())
+    EMBED_DIM = 32
+    NUN_CLASS = len(train_dataset.get_labels())
```

Comment:
maybe use `LightningDataModule` for this stuff?

Reply:
+1 on this. @nahidgrid reach out if ya want help on doin that.